src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -503,4 +507,104 @@ public void replaceVariablesInFile(String filePath, List<String> targetStrings,
         }
         Files.write(replaceFilePath, fileContent.getBytes(charset));
     }
+
+    /**
+     * This normalizes all line endings to UNIX-line-endings recursively from the startPath.
+     *
+     * @see {@link #normalizeLineEndings(String) normalizeLineEndings}
+     * @param startPath          the path where the start directory is located
+     * @throws IOException if an issue occurs on file access for the normalizing of the line endings.
+     */
+    public void normalizeLineEndingsRecursive(String startPath) throws IOException {
```

Comment:
I think the name is misleading, as this implementation does not utilize recursion.

Reply:
This was a remnant of an earlier implementation, I changed it to be more specific.